Fix AnimationPlayer `blend_times` sorting

JacobMillner opened the pull request:
Fixes #76491 

The current operator< for `BlendKeys` compares to/from pointers resulting in nondeterministic behavior. You can see this represented in the `.tscn` file of anything with lots of `blend_times` resulting in file changes which leads to extra noise in your git commits.

This PR uses`StringName::AlphaCompare` when making to/from comparisons in order preform a lexicographic order sort rather than one determined by memory location.

AThousandShips commented on scene/animation/animation_player.h:
```diff
@@ -94,10 +94,10 @@ class AnimationPlayer : public AnimationMixer {
 		bool operator<(const BlendKey &bk) const {
-			if (from == bk.from) {
-				return to < bk.to;
+			if (String(from) == String(bk.from)) {
```
Why this? There's no reason to compare them as strings, that's significantly slower

JacobMillner replied:
Good point and thanks for the review, I will change this back to `if (from == bk.from)`. Testing it locally seems to work fine and not result in sort variations.

Do you know if comparing two `StringName`s  that have the same string literal will always point to the same mem address?

AThousandShips replied:
You should just use `StringName::AlphCompare` here, like so:
```cpp
return StringName::AlphCompare()(from, bk.from) < 0;
```

AThousandShips commented on scene/animation/animation_player.h:
```diff
@@ -95,9 +95,9 @@ class AnimationPlayer : public AnimationMixer {
 		bool operator<(const BlendKey &bk) const {
 			if (from == bk.from) {
-				return to < bk.to;
+				return String(to) < String(bk.to);
```
```suggestion
				return StringName::AlphCompare()(to, bk.to) < 0;
```

AThousandShips commented on scene/animation/animation_player.h:
```diff
@@ -95,9 +95,9 @@ class AnimationPlayer : public AnimationMixer {
+				return String(to) < String(bk.to);
 			} else {
-				return from < bk.from;
+				return String(from) < String(bk.from);
```
```suggestion
				return StringName::AlphCompare()(from, bk.from) < 0;
```

AThousandShips commented on scene/animation/animation_player.h:
```diff
@@ -95,9 +95,9 @@ class AnimationPlayer : public AnimationMixer {
 		bool operator<(const BlendKey &bk) const {
 			if (from == bk.from) {
-				return to < bk.to;
+				return StringName::AlphCompare()(to, bk.to) < 0;
```
```suggestion
				return StringName::AlphCompare()(to, bk.to);
```
Forgot it was a `<` not a ordering return

JacobMillner commented:
tested and working locally with `StringName::AlphaCompare` :+1:

TokageItLab commented:
Please squash the commits to follow PR workflow https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#the-interactive-rebase

JacobMillner committed:
fix the sorting of blend_times in animation player

Modify BlendKey's sort to use  AlphaCompare in order to create a deterministic sort

Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

akien-mga commented:
Thanks!

JacobMillner commented:
Thanks everyone!
